services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/PortPropertiesSpec.groovy

```diff
@@ -0,0 +1,251 @@
+    }
+
+    @Tags(VIRTUAL)
+    def "System doesn't discover link when port discovery property is disabled"() {
+        given: "A deleted link"
+        def sw = topology.activeSwitches.first()
+        def relatedIsls = topology.getRelatedIsls(sw)
+        def islToManipulate = relatedIsls.first()
+
+        // Bring port down on the src switch
+        antiflap.portDown(islToManipulate.srcSwitch.dpId, islToManipulate.srcPort)
+        Wrappers.wait(discoveryTimeout + WAIT_OFFSET) {
+            islUtils.getIslInfo(islToManipulate).get().state == IslChangeType.FAILED
+        }
+
+        // delete link
+        northbound.deleteLink(islUtils.toLinkParameters(islToManipulate))
+
+        when: "Disable port discovery property on the src and dst switches"
+        northboundV2.updatePortProperties(islToManipulate.srcSwitch.dpId, islToManipulate.srcPort,
+                new PortPropertiesDto(discoveryEnabled: false))
+        northboundV2.updatePortProperties(islToManipulate.dstSwitch.dpId, islToManipulate.dstPort,
+                new PortPropertiesDto(discoveryEnabled: false))
+
+        and: "Bring port up on the src switch"
+        antiflap.portUp(islToManipulate.srcSwitch.dpId, islToManipulate.srcPort)
+
+        then: "Link is not detected"
+        Wrappers.timedLoop(discoveryInterval + WAIT_OFFSET / 2) {
+            assert !islUtils.getIslInfo(islToManipulate).isPresent()
+        }
+
+        when: "Deactivate/activate src switch"
+        lockKeeper.knockoutSwitch(sw)
+        Wrappers.wait(discoveryTimeout + WAIT_OFFSET) {
+            assert northbound.getSwitch(sw.dpId).state == SwitchChangeType.DEACTIVATED
+            assert northbound.getAllLinks().findAll {
+                it.state == IslChangeType.FAILED
+            }.size() == (relatedIsls - islToManipulate).size() * 2
+        }
+
+        lockKeeper.reviveSwitch(sw)
+        Wrappers.wait(discoveryInterval + WAIT_OFFSET) {
+            assert northbound.getSwitch(sw.dpId).state == SwitchChangeType.ACTIVATED
+            def links = northbound.getAllLinks()
+            (relatedIsls - islToManipulate).each {
+                assert islUtils.getIslInfo(links, it).get().state == IslChangeType.DISCOVERED
+            }
+        }
+
+        then: "Link is still not detected"
+        Wrappers.timedLoop(discoveryInterval + WAIT_OFFSET / 2) {
+            assert !islUtils.getIslInfo(islToManipulate).isPresent()
+        }
+
+        when: "Enable port discovery property on the src switch"
+        northboundV2.updatePortProperties(islToManipulate.srcSwitch.dpId, islToManipulate.srcPort,
+                new PortPropertiesDto(discoveryEnabled: true))
+
+        then: "Link is detected and link status is FAILED"
+        Wrappers.wait(discoveryInterval + WAIT_OFFSET) {
+            def islInfoForward = islUtils.getIslInfo(islToManipulate).get()
+            def islInfoReverse = islUtils.getIslInfo(islToManipulate.reversed).get()
```

Comment:
you are dumping all isls twice here. Actually not a big deal, but a cleaner approach would be to use
`northbound.getLink(islToManipulate)` to get only 1 link instead of all OR user the alternative version of getIslInfo that don't query northbound for links
```
def allLinks = northbound.getAllLinks()
def islInfoForward = islUtils.getIslInfo(allLinks, islToManipulate).get()
def islInfoReverse = islUtils.getIslInfo(allLinks, islToManipulate.reversed).get()

```

Reply:
thx, fixed